device/cluster.cpp

```diff
@@ -2932,6 +2973,16 @@ void Cluster::l1_membar(
     }
 }
 
+void Cluster::l1_membar(
+    const chip_id_t chip, const std::unordered_set<tt::umd::CoreCoord>& cores, const std::string& fallback_tlb) {
```

Comment:
Is there a way to use `std::ranges::input_range` instead of `std::unordered_set<>`?

Reply:
I can investigate this, not as part of this PR, this is just for API. Follow up issue https://github.com/tenstorrent/tt-umd/issues/427